api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/AbstractConv.kt

```diff
@@ -100,14 +132,50 @@ public abstract class AbstractConv(
     override val paramCount: Int
         get() = (kernelShape.numElements() + biasShape.numElements()).toInt()
 
+    /** Define the number of output channels given the number of input channels.
+     *  Defaults to the number of filter in convolutional layer. */
+    protected open fun getOutputDepth(numberOfChannels: Long): Long = filtersInternal
+
+    /**
+     * Define the [kernelShape] and [biasShape] by default from its [kernelSizeInternal],
+     * [filtersInternal], [filtersInternal] and the given [numberOfChannels] from input Tensor.
+     *
+     * @param numberOfChannels for input of this layer
+     */
+    protected open fun computeMatricesShapes(numberOfChannels: Long) {
```

Comment:
I suggest splitting it into two small functions for each shape calculation, also rename it to something clear instead of computeMatricesShapes, more related to kernel and bias

Reply:
done